### PR TITLE
GSYE-425: Fix bug due to merge of refactoring branch

### PR DESCRIPTION
**Pull request by hannesdiedrich**
## Reason for the proposed changes

Please describe what we want to achieve and why.

## Proposed changes

-

**INTEGRATION_TESTS_BRANCH**=master
**GSY_FRAMEWORK_BRANCH**=master

### CODE REVIEW

File: src/gsy_e/models/area/coefficient_area.py [hunk 1]
```diff
@@ -36,16 +36,18 @@ class CoefficientAreaException(Exception):
...
 
-def assert_coefficient_area_settings(
-        strategy: Optional[SCMStrategy], setting: Optional[float], setting_name: str) -> float:
+def validate_coefficient_area_setting(
+        children: Optional[List], setting: Optional[float], setting_name: str) -> float:
     """Check if coefficient area that is not an asset provided SCM setting."""
-    if strategy is None and setting is None:
-        raise CoefficientAreaException(f"In SCM simulations {setting_name} can not be None.")
+    if children:
+        if all(not child.children for child in children) and setting is None:
```
Comment (spyrostz) [inline src/gsy_e/models/area/coefficient_area.py:43]: As far as I see there is a method called `_is_home_area` in the CoefficientArea class, can it not be used here too ?

**Review by spyrostz — APPROVED** [review]
LGTM already, one possible enhancement comment

File: src/gsy_e/models/area/coefficient_area.py (hunk 1 above)
Comment (hannesdiedrich) [reply src/gsy_e/models/area/coefficient_area.py]: You were mentioning this in the original PR already IIRC, but I must have ignored it. Sorry.
I adapted it here:[0bfff14](https://github.com/gridsingularity/gsy-e/pull/1551/commits/0bfff1456706bc89ea43f2019b71e597d5457d43)
I realised, that I can also move the check for the grid_fees before the init of the super class. Thanks a lot!

**Review by spyrostz — APPROVED** [review]
LGTM

**Review by mrsaemir — APPROVED** [review]

**Pull request merged**